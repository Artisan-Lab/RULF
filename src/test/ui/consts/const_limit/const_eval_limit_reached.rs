#![feature(const_eval_limit)]
#![const_eval_limit = "500"]

const X: usize = {
    let mut x = 0;
    while x != 1000 {
        //~^ ERROR evaluation of constant value failed
        x += 1;
    }

    x
};

fn main() {
    assert_eq!(X, 1000);
}
