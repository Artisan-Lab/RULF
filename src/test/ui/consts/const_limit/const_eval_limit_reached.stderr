error[E0080]: evaluation of constant value failed
  --> $DIR/const_eval_limit_reached.rs:6:11
   |
LL |     while x != 1000 {
   |           ^^^^^^^^^ exceeded interpreter step limit (see `#[const_eval_limit]`)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0080`.
